src/main/webapp/js/webrtc_adaptor.js

```diff
@@ -763,19 +763,19 @@ function WebRTCAdaptor(initialValues)
 			console.log("Data channel is closed");
 			thiz.callback("data_channel_closed", streamId);
 		};
-		
+
 	}
 
 
-	this.initPeerConnection = function(streamId) {
-		if (thiz.remotePeerConnection[streamId] == null) 
+	this.initPeerConnection = function(streamId, forPublishing) {
```

Comment:
forPublishing parameter seems a little confusing for me. We give this parameter as true in several places. 
Why do we need such a parameter?

Reply:
In the method takeConfiguration initPeerConnection is called with forPublishing=false if the typeConfiguration is "offer". These is the case where you play the streams of other conference participants. DataChannel is created in their side for these streams.
In this case no DataChannel is created on our side exxplicitly but it is assigned in onDataChannel method. 
The naming of the parameter forPublishing is bad, probably something like createDataChannel would be better.

Reply:
I see then I just some comments below.